app/controllers/responses_controller.rb

```diff
@@ -27,16 +28,27 @@ def update
     @response.update(response_params)
     respond_to do |format|
       format.json { render json: @response.to_json }
-    end  end
+    end
+  end
 
   def destroy
     @response.destroy
     redirect_to discussions_path(discussion_id: @discussion.id)
   end
 
+  def upvote
+    if current_user
+      @response = Response.find(params[:response_id])
+      @response.upvotes.create(user_id: current_user.id)
+      @response.score += 1
```

Comment:
Same thing as discussions controller += error: 
response.score is default "nil"so add a check for "if @response.score" to do += 1, else just set it equal to 1